src/truflation/data/exporter.py

```diff
@@ -88,8 +88,30 @@ def export(self, export_details: ExportDetails, df_local: pandas.DataFrame, dry_
                     df_new_data
                 )
 
+            # Ensure primary key is set
+            self.ensure_primary_key(export_details, df_new_data)
+
         return df_new_data
 
+    @staticmethod
+    def ensure_primary_key(export_details: ExportDetails, df_new_data: pandas.DataFrame) -> None:
+        """
+        Ensures that the primary key is created if they do not exist.
+
+        param:
+          export_details: ExportDetails: database details
+          df_new_data: Pandas.DataFrame: dataframe to export
+        """
+        engine = export_details.writer
+        result = engine.execute([f"SHOW INDEX FROM {export_details.key} WHERE Key_name = 'PRIMARY'"])
```

Comment:
Also, not sure about using `key` as the value of the table name.

Please let me know

Reply:
The default primary keys should be date/created_at.

This code is very messy because it hard codes date as the primary key which breaks with tables that don't have date, but I think the best thing to do is to default primary key to date and created_at, but make the system work without saving primary keys if date/created_at is not available.

For regression tests, you should save a table that has date/created_at to see if the primary key works, and then a table that doesn't have either item to make sure it can fall back.

Reply:
Hey @joequant my code is based on the facts that:

- Primary key should guarantee the uniqueness
- using only `date` and `created_at` doesn't guarantee that e.g.
	<img width="1418" alt="image" src="https://github.com/user-attachments/assets/f593f865-f819-4e56-afd4-f62224b5952d">
- using the primary key on all the table fields ensures the primary key is unique

Please let me know what you think.

cc. @Jarryd-pretorius, @ambahk

Reply:
Yeah, for categories the unique keys are data, category, subcategory, and created_date.

Reply:
> Yeah, for categories the unique keys are data, category, subcategory, and created_date

So there needs to be a way of setting exports so that it can specify primary keys and default to date, created_at.  For the ingestion tables it's just date value, for the other tables it could include additional data.  This will also be true for normalized_data and the argentina government table.

Reply:
@joequant, I want to double-check why my version with `all fields` as PK is not an option.

Reply:
Hmmmm....  Setting the primary keys to everything.  Yeah that might be an option, but just make sure that you don't run into some odd performance issues.

Reply:
@joequant Please review this PR.
I also added a method for updating the `created_at` field as the table `edu_psu_housing` has a lot of NULL values.